react/features/conference/components/native/NavigationBar.js

```diff
@@ -0,0 +1,87 @@
+// @flow
+
+import _ from 'lodash';
+import React, { Component } from 'react';
+import { SafeAreaView, Text, View } from 'react-native';
+import LinearGradient from 'react-native-linear-gradient';
+import { connect } from 'react-redux';
+
+import { PictureInPictureButton } from '../../../mobile/picture-in-picture';
+import { isToolboxVisible } from '../../../toolbox';
+
+import styles, { NAVBAR_GRADIENT_COLORS } from './styles';
+
+type Props = {
+
+    /**
+     * Name of the meeting we're currently in.
+     */
+    _meetingName: string,
+
+    /**
+     * True if the navigation bar should be visible.
+     */
+    _visible: boolean
+};
+
+/**
+ * Implements a navigation bar component that is rendered on top of the
+ * conference screen.
+ */
+class NavigationBar extends Component<Props> {
+    /**
+     * Implements {@Component#render}.
+     *
+     * @inheritdoc
+     */
+    render() {
+        if (!this.props._visible) {
+            return null;
+        }
+
+        return (
+            <View style = { styles.navBarContainer }>
+                <LinearGradient
+                    colors = { NAVBAR_GRADIENT_COLORS }
+                    style = { styles.gradient } >
+                    <SafeAreaView style = { styles.navBarSafeView }>
+                        <View style = { styles.navBarWrapper }>
+                            <PictureInPictureButton
+                                styles = { styles.navBarButton } />
+                            <View style = { styles.roomNameWrapper }>
+                                <Text style = { styles.roomName }>
+                                    { this.props._meetingName }
+                                </Text>
+                            </View>
+                        </View>
+                    </SafeAreaView>
+                </LinearGradient>
+            </View>
+        );
+    }
+
+}
+
+/**
+ * Maps part of the Redux store to the props of this component.
+ *
+ * @param {Object} state - The Redux state.
+ * @returns {{
+ *     _meetingName: string,
+ *     _visible: boolean
+ * }}
+ */
+function _mapStateToProps(state) {
+    const { callee } = state['features/base/jwt'];
+    const meetingName
+            = state['features/base/config'].callDisplayName
+                || (callee && callee.name)
+                || state['features/base/conference'].room;
+
+    return {
+        _meetingName: _.startCase(meetingName),
+        _visible: isToolboxVisible(state)
```

Comment:
Same. This relationship is spurious. I'd say we add a new action which we dispatch at the same time as we dispatch the toolbox visibility one.

Reply:
I disagree here. If you check the code of ```isToolboxVisible``` you'll see that its return value depends on multiple parameters. We could of course implement some action that imitates its behaviour, or even fully implements it, but it would be a rather complicated logic (set an initial value on app start based on ```alwaysVisible``` and ```enabled```, then listen to the changes of ```visible``` runtime) then we'd still have the same dependency, except that we don't invoke a function from toolbox, but listen to actions from toolbox. Same-same, you know, but this is easier to understand 1 year from now.

On the other hand we may think of something more generic, e.g. create a separate (perhaps ```base```) feature called ```controls``` and make the visibility of toolbox, navbar, labels and all others depend on it, since we came a long way from the state where we only had toolbar and nothing else. And I'm quite certain we'll have other features depending on its state pretty soon.

I'm happy to do it if you like the idea, but I'd insist to stay away from simply turning this particular function into a set of Redux actions.

Reply:
I didn't mean to replicate the exact same behavior, but to control its visibility the same way: when the user taps the screen.

I don't want to see this 2 components coupled by the state because they are independent. You can pass a visible prop too, like the circular labels.